### ISSUE TITLE
H2ONotFoundArgumentException when trying save trained model with ModelSerializationSupport.exportMOJOModel

### ISSUE THREAD

**Issue by hangelwen** [h2oai/sparkling-water#732]
The following code triggered H2ONotFoundArgumentException:
This code
```
ModelSerializationSupport.exportMOJOModel(gbmModel, new URI("hdfs:/path/to/gbmModel.zip"))
```
triggered H2ONotFoundArgumentException:

```
Exception in thread "main" water.exceptions.H2ONotFoundArgumentException: Failed to find schema for version: 3 and type: GBMModel
        at water.api.SchemaServer.schema(SchemaServer.java:282)
        at water.api.SchemaServer.schema(SchemaServer.java:252)
        at hex.ModelMojoWriter.writeModelDetails(ModelMojoWriter.java:277)
        at hex.ModelMojoWriter.writeTo(ModelMojoWriter.java:178)
        at hex.ModelMojoWriter.writeTo(ModelMojoWriter.java:169)
        at hex.ModelMojoWriter.writeTo(ModelMojoWriter.java:161)
        at hex.Model.exportMojo(Model.java:2300)
        at water.support.ModelSerializationSupport$class.exportMOJOModel(ModelSerializationSupport.scala:60)
        at water.support.ModelSerializationSupport$.exportMOJOModel(ModelSerializationSupport.scala:73)
```

Sparkling Water I tried includes 2.0.27 and 2.3.1

**Comment by jakubhava** [comment]
Can you please share full and minimal reproducible code?

**Comment by hangelwen** [comment]
@jakubhava It's just a piece of code to train a GBM model and then call this method to save the model... Removing this step the code runs fine. Also this line of code worked in Zeppelin. But when I built a jar and submit it to the cluster it failed. 


```
val data = spark.read.parquet(datapath).cache
val h2oContext = H2OContext.getOrCreate(spark)
import h2oContext._
import h2oContext.implicits._
import spark.implicits._
val trainingData = h2oContext.asH2OFrame(data, "training-data-set")
val gbmParams = new GBMParameters()
gbmParams._train = trainingData
gbmParams._response_column = "target"
gbmParams._ntrees = 3
gbmParams._max_depth = 5
val gbmname ="gbmModel.hex"
val gbm= new GBM(gbmParams, Key.make(gbmname))
val gbmModel = gbm.trainModel().get()
ModelSerializationSupport.exportMOJOModel(gbmModel, new URI("hdfs:/path/to/gbmModel.zip"))

```

**Comment by hangelwen** [comment]
@jakubhava 
Actually while there was a H2ONotFoundArgumentException,  there was a file saved at the given HDFS path, but the file size looked small and is not a valid zip file. When I tried to unzip I got the following error:
```
Archive:  glmModel.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of glmModel.zip or
        glmModel.zip.zip, and cannot find glmModel.zip.ZIP, period.
```

**Comment by hangelwen** [comment]
@jakubhava a similar mode saved using the same code in Zeppelin.
```
Archive:  gbmModel.zip
Zip file size: 44992 bytes, number of entries: 12
-rw----     2.0 fat      688 bl defN 18-Jun-01 07:02 trees/t00_000.bin
-rw----     2.0 fat     1160 bl defN 18-Jun-01 07:02 trees/t00_000_aux.bin
-rw----     2.0 fat      689 bl defN 18-Jun-01 07:02 trees/t00_001.bin
-rw----     2.0 fat     1080 bl defN 18-Jun-01 07:02 trees/t00_001_aux.bin
-rw----     2.0 fat      775 bl defN 18-Jun-01 07:02 trees/t00_002.bin
-rw----     2.0 fat     1200 bl defN 18-Jun-01 07:02 trees/t00_002_aux.bin
-rw----     2.0 fat      766 bl defN 18-Jun-01 07:02 model.ini
-rw----     2.0 fat      119 bl defN 18-Jun-01 07:02 domains/d000.txt
-rw----     2.0 fat     2985 bl defN 18-Jun-01 07:02 domains/d001.txt
-rw----     2.0 fat        4 bl defN 18-Jun-01 07:02 domains/d002.txt
-rw----     2.0 fat   125898 bl defN 18-Jun-01 07:02 experimental/modelDetails.json
-rw----     2.0 fat      164 bl defN 18-Jun-01 07:02 experimental/README.md
12 files, 135528 bytes uncompressed, 43420 bytes compressed:  68.0%
```

**Comment by hangelwen** [comment]
@jakubhava Same issue when I tried to save to local path: a small invalid zip file was created with same exception.

**Comment by jakubhava** [comment]
@hangelwen are you building sparkling water or using official distribution? Can you please try your code on official distribution which you can download from https://www.h2o.ai/download/

**Comment by hangelwen** [comment]
@jakubhava Using `--jar` to provide the dependency solved the issue (at least for the 2.1.30 SW version I tested)

**Comment by WendyYW** [comment]
> @jakubhava Using `--jar` to provide the dependency solved the issue (at least for the 2.1.30 SW version I tested)

eh, you use `--jar` when compiling h2o or somewhere?

**Comment by WendyYW** [comment]
> @jakubhava Using `--jar` to provide the dependency solved the issue (at least for the 2.1.30 SW version I tested)

eh, you use --jar when compiling h2o or somewhere? I met the similiar issue when using it
